sqlmesh/core/snapshot/evaluator.py

```diff
@@ -1098,7 +1098,15 @@ def migrate(
         )
 
     def delete(self, name: str) -> None:
-        self.adapter.drop_view(name)
+        try:
+            self.adapter.drop_view(name)
+        except Exception:
+            logger.debug(
+                "Failed to drop view '%s'. Trying to drop the materialized view instead",
+                name,
+                exc_info=True,
+            )
```

Comment:
We don't have a way of knowing if the view was materialized, so we need to use a brute force approach?

Reply:
Yes, unfortunately. This information is not available when we're cleaning things up.